File: basic/src/main/java/com/yuanshenbin/basic/base/BaseListMvpFragment.java
package com.yuanshenbin.basic.base;

import com.aspsine.swipetoloadlayout.SwipeToLoadLayout;
import com.yuanshenbin.basic.adapter.CommonAdapter;
import com.yuanshenbin.basic.constant.BasicConstants;


import java.util.ArrayList;
import java.util.List;

/**
 * author : yuanshenbin
 * time   : 2018/7/25
 * desc   :
 */

public abstract class BaseListMvpFragment<VH extends BasicViewHolder, Bean, V, P extends BasePresenter<V>>
        extends BaseMvpFragment<VH, V, P> {
    protected SwipeToLoadLayout swipe_to_load_layout;
    protected List<Bean> mData = new ArrayList<>();
    protected CommonAdapter<Bean> mAdapter;

    public void handleListData(Object result, boolean pull, int page,int loading) {
        handleListData(result, pull, page, BasicConstants.PAGESIZE,loading);
    }

    public void handleListData(Object result, boolean pull, int page, int size,int loading) {
        swipe_to_load_layout.setRefreshing(false);
        if (mDelegate != null) {
            mDelegate.handleListData(mAdapter, mData, result, pull, page, size,loading);
        }
    }

    protected void setListLoadFail() {
        swipe_to_load_layout.setRefreshing(false);
        if (!isPullAndPush) {
            mPage--;
            mAdapter.loadMoreFail();
        }
    }
}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/base/BaseListMvpFragment.kt
package com.yuanshenbin.basic.base

import com.aspsine.swipetoloadlayout.SwipeToLoadLayout
import com.yuanshenbin.basic.adapter.CommonAdapter
import com.yuanshenbin.basic.constant.BasicConstants
import java.util.*

/**
 * author : yuanshenbin
 * time   : 2018/7/25
 * desc   :
 */
abstract class BaseListMvpFragment<VH : BasicViewHolder?, Bean, V, P : BasePresenter<V>?> : BaseMvpFragment<VH, V, P>() {
    @JvmField
    var swipe_to_load_layout: SwipeToLoadLayout? = null

    @JvmField
    var mData: List<Bean?> = ArrayList()

    @JvmField
    var mAdapter: CommonAdapter<Bean?>? = null
    fun setFill(result: Any?, loading: Int, page: Int, size: Int, emptyMsg: CharSequence?) {
        swipe_to_load_layout!!.isRefreshing = false
        mDelegate!!.handleListData(mAdapter, mData, result, isPullAndPush, loading, page, size, emptyMsg)
    }

    fun setFill(result: Any?, loading: Int, size: Int) {
        setFill(result, loading, mPage, size, null)
    }

    fun setFill(result: Any?, loading: Int) {
        setFill(result, loading, BasicConstants.PAGESIZE)
    }

    fun setFill(result: Any?, loading: Int, empty: CharSequence?) {
        setFill(result, loading, mPage, BasicConstants.PAGESIZE, empty)
    }

    protected fun setListLoadFail() {
        swipe_to_load_layout!!.isRefreshing = false
        if (!isPullAndPush) {
            mPage--
            if (mAdapter!!.loadMoreModule != null) {
                mAdapter!!.loadMoreModule.loadMoreFail()
            }
        }
    }
}